src/app/pages/storage/stores/pools-dashboard-store.service.ts

```diff
@@ -107,11 +106,7 @@ export class PoolsDashboardStore extends ComponentStore<PoolsDashboardState> {
   }
 
   getPools(): Observable<Pool[]> {
-    return this.ws.call('pool.query', [[], { extra: { is_upgraded: true } }]);
-  }
-
-  listenForPoolUpdates(): Observable<ApiEvent<Pool>> {
-    return this.ws.subscribe('pool.query');
+    return this.ws.callAndSubscribe('pool.query', [[], { extra: { is_upgraded: true } }]);
```

Comment:
Should we handle subscriptions `untilDestroyed` or smth?
For cases when I leave `Pools Dashboard` page and then return back?

Reply:
Good point. I assumed that it managed by store effect and there is general protection in web socket to unsubscribe when session is recreated. I'll double check it.

Reply:
The `PoolsDashboardStore` instance is part of the `StorageModule` and remains active until the tab is closed. The primary function, `loadDashboard()`, is called from various locations, including when navigating to the Pools Dashboard. As such, it calls `pool.query` and other endpoints and then restores the "subscribe" observable (API event updates) from subscriptions in the websocket service. In this case, we do not need to unsubscribe.

It is unclear whether the `loadDashboard()` method unsubscribes from previous subscriptions or if the `PoolsDashboardStore` architecture has any memory leaks. These are excellent questions that require additional investigation, but unfortunately, they are beyond the scope of this ticket. Feel free to log a ticket for further investigation.

Reply:
@undsoft - please check these comments as well

Reply:
You need to unsubscribe where you subscribe (as in rxjs subscribe). This happens in the consuming component.
As long as `callAndSubscribe` stops listening to middleware events when last subscriber unsubscribes, this is okay.

Reply:
Maybe we should replaced the "subscribe" keyword for these method names with "getEvents" or "subscribeToEvents" or something more declarative so we can avoid confusions like this in the future. Can't say this makes a very strong case for renaming though. Just a thought.

Reply:
This name directly states what the method does, which is first to make a `call` and then to `subscribe` to updates. In context of websocket service those are known terms. I'd prefer to keep it short and clear.